### ISSUE TITLE
Token authentication is broken

### ISSUE THREAD

**Issue by ng-galien** [spring-projects/spring-pulsar#529]
Since `org.apache.pulsar.client.impl.auth.AuthenticationToken` does not handle `configure(Map<String, String> authParams)`

We cannot use this plugin as it's [documented](https://docs.spring.io/spring-pulsar/docs/1.0.0/reference/reference/pulsar.html#pulsar-client)

```yaml
spring:
  pulsar:
    client:
      authentication:
        plugin-class-name: org.apache.pulsar.client.impl.auth.AuthenticationToken
        param:
          token: some-token-goes-here
```

```
 WARN- 23-12-20 19:59:39.966 1351900-[r-client-io-1-5] o.a.p.c.i.ClientCnx                       : [localhost/127.0.0.1:6650] Got exception java.lang.RuntimeException: failed to get client token
	at org.apache.pulsar.client.impl.auth.AuthenticationDataToken.getToken(AuthenticationDataToken.java:64)
	at org.apache.pulsar.client.impl.auth.AuthenticationDataToken.<init>(AuthenticationDataToken.java:36)
```

The doc must be updated and or parameters injection must be adjusted.

Also reported [in this issue ](https://github.com/apache/pulsar/issues/21770) on Pulsar repo.

**Comment by onobc** [comment]
Hi @ng-galien this has been fixed in https://github.com/spring-projects/spring-pulsar/issues/500 and will be available w/ the release of Spring Boot 3.2.1 tomorrow (2023-12-21).

The auto-configuration piece is in Spring Boot and was updated https://github.com/spring-projects/spring-boot/pull/38839

**Comment by ng-galien** [comment]
@onobc great!
